projects/ngneat/transloco/src/lib/transloco.pipe.ts

```diff
@@ -77,4 +74,13 @@ export class TranslocoPipe implements PipeTransform, OnDestroy {
     this.lastValue = this.translocoService.translate(key, params, lang);
     this.cdr.markForCheck();
   }
+
+  private resolveScope(lang: string, providerScope: string | ProviderScope | null) {
```

Comment:
Do you want me to put this in the service? I would have to pass a lot of parameters (scopeResolver, langResolver,  if I were to move that piece of code to the service, is that okay?

Reply:
@itayod

Reply:
@itayod If I move this function to the transloco service I wouldn't be able to set `this.path` which is needed by `updateValue()` because it will be out of scope when moved to the service. Any suggestions?

Reply:
it's ok, leave it the way it is now :)